lib/src/misc/simplify.dart

```diff
@@ -0,0 +1,120 @@
+// implementation based on
```

Comment:
(referring to the file not the line) Is there a way we can make this simplification logic available more accessible publicly (as @ignatz), ie. in a single method?
Or did we want to keep this private for some reason?

Reply:
no reason to keep it private! it is just an implementation of simplify.js but with flutter_map typings to make things nice. we might keep it private to allow us to change the types in the future, and there might be simplify.js implementations for dart on pub, so i think we can just keep it private.

Reply:
Ok, I see what you're saying, we don't want to confuse FM's scope as well. But I'm still not sure hiding it is the best either (if we've already got the functionality, why make someone install another package) - but I'll leave it up to you.

In any case, I think it should be refactored into a class called something like `Simplify`, and the `simplify` method either made static or renamed to `call`, and the two roots of the algorithms made public within the class. No specific reason exactly, apart from that I personally dislike trailing functions like this, especially when there's many in a file that all have related/interlinked functions - bundling them up just feels nicer. Not sure if there's any point having a constructor or any non-statics though.

Reply:
i moved simplify into a top level library. not sure if this is actually the best way to do it. im hesitant to wrap it in a class with all static members just for the sake of doing that. it's actually a linting rule in dart to avoid it and it is part of effective dart: https://dart.dev/effective-dart/design#avoid-defining-a-class-that-contains-only-static-members

see latest commit for the change